assets/js/base/components/button/index.tsx

```diff
@@ -2,27 +2,29 @@
  * External dependencies
  */
 import { Button as WPButton } from 'wordpress-components';
-import PropTypes from 'prop-types';
+import type { ReactNode } from 'react';
 import classNames from 'classnames';
-
 /**
  * Internal dependencies
  */
 import './style.scss';
 
-/** @typedef {import('react')} React */
+interface ButtonProps extends WPButton.ButtonProps {
+	className?: string;
+	showSpinner?: boolean;
+	children?: ReactNode;
+}
 
 /**
  * Component that visually renders a button but semantically might be `<button>` or `<a>` depending
  * on the props.
- *
- * @param {Object}              props             Incoming props for the component.
- * @param {string}              props.className   CSS classname used.
- * @param {boolean}             props.showSpinner Whether to show spinner or not.
- * @param {React.ReactChildren} props.children    Child components passed in.
- * @param {Object}              props.props       Rest of incoming props.
  */
-const Button = ( { className, showSpinner = false, children, ...props } ) => {
+const Button = ( {
+	className,
+	showSpinner = false,
+	children,
+	...props
+}: ButtonProps ) => {
```

Comment:
Not sure why in the other PR I see an ESLint warning in this line while in this PR there isn't. :confused: 

https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/4098/files#diff-3d98a12115787e07b0261408773117bb47a24318d3165adc24a0729ac3846133R27

Reply:
Yeah this is strange, it never shows in my IDE either. I've fixed it now anyway :)